core/src/main/java/org/geysermc/geyser/command/CommandRegistry.java

```diff
@@ -0,0 +1,333 @@
+        registerBuiltInCommand(new AdvancedTooltipsCommand("tooltips", "geyser.commands.advancedtooltips.desc", "geyser.command.tooltips"));
+        registerBuiltInCommand(new ConnectionTestCommand(geyser, "connectiontest", "geyser.commands.connectiontest.desc", "geyser.command.connectiontest"));
+        if (this.geyser.getPlatformType() == PlatformType.STANDALONE) {
+            registerBuiltInCommand(new StopCommand(geyser, "stop", "geyser.commands.stop.desc", "geyser.command.stop"));
+        }
+
+        if (!this.geyser.extensionManager().extensions().isEmpty()) {
+            registerBuiltInCommand(new ExtensionsCommand(this.geyser, "extensions", "geyser.commands.extensions.desc", "geyser.command.extensions"));
+        }
+
+        GeyserDefineCommandsEvent defineCommandsEvent = new GeyserDefineCommandsEventImpl(this.commands) {
+
+            @Override
+            public void register(@NonNull Command command) {
+                if (!(command instanceof GeyserExtensionCommand extensionCommand)) {
+                    throw new IllegalArgumentException("Expected GeyserExtensionCommand as part of command registration but got " + command + "! Did you use the Command builder properly?");
+                }
+
+                registerExtensionCommand(extensionCommand.extension(), extensionCommand);
+            }
+        };
+
+        this.geyser.eventBus().fire(defineCommandsEvent);
+
+        for (Map.Entry<Extension, Map<String, Command>> entry : this.extensionCommands.entrySet()) {
+            Extension extension = entry.getKey();
+
+            // Register this extension's root command
+            extensionRootCommands.put(extension.rootCommand(), extension);
+
+            // Register help commands for all extensions with commands
+            String id = extension.description().id();
+            registerExtensionCommand(extension, new HelpCommand(
+                this.geyser,
+                "help",
+                "geyser.commands.exthelp.desc",
+                "geyser.command.exthelp." + id,
+                extension.rootCommand(),
+                entry.getValue()));
+        }
+
+        // wait for the right moment (depends on the platform) to register permissions
+        geyser.eventBus().subscribe(new GeyserEventRegistrar(this), GeyserRegisterPermissionsEvent.class, this::onRegisterPermissions);
+    }
+
+    @NotNull
+    public CommandManager<GeyserCommandSource> cloud() {
+        return cloud;
+    }
+
+    @NotNull
+    public Map<String, Command> commands() {
+        return Collections.unmodifiableMap(this.commands);
+    }
+
+    @NotNull
+    public Map<Extension, Map<String, Command>> extensionCommands() {
+        return Collections.unmodifiableMap(this.extensionCommands);
+    }
+
+    public void clear() {
+        this.commands.clear();
+        this.extensionCommands.clear();
+        this.permissionDefaults.clear();
+    }
```

Comment:
Why are `extensionRootCommands` not cleared?

Reply:
missed it

Just realized that `#clear()` may have to be removed though, or handled by the bootstrap (probably more cohesive anyway), since some bootstraps only load commands once